haystack/database/elasticsearch.py

```diff
@@ -143,6 +143,10 @@ def write_documents(self, documents: List[dict]):
             documents_to_index.append(_doc)
         bulk(self.client, documents_to_index, request_timeout=300)
 
+    def update_document_meta(self, id: str, meta: Dict[str, str]):
+        body = {"doc": meta}
```

Comment:
Wouldn't that give us a nested structure here (different to the flat one we use when indexing a doc for the first time)?

Reply:
It should retain the flat structure. All individual fields inside `meta` get updated.

Reply:
Can you add a small test for it, please :)